app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java

```diff
@@ -157,6 +178,66 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        try {
+            // When an Image is picked
+            if (requestCode == 1 && resultCode == RESULT_OK
+                    && null != data) {
+                // Get the Image from data
+
+                Uri selectedImage = data.getData();
+                String[] filePathColumn = { MediaStore.Images.Media.DATA };
+
+                // Get the cursor
+                Cursor cursor = getActivity().getApplicationContext().getContentResolver().query(
+                        selectedImage,
+                        filePathColumn, null, null, null);
+                // Move to first row
+                cursor.moveToFirst();
+
+                int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+                imgDecodableString = cursor.getString(columnIndex);
+                cursor.close();
+                // Set the Image in ImageView after decoding the String
+                ivProfile.setVisibility(View.GONE);
+                ivTextDrawable.setVisibility(View.GONE);
+                iv_profilepic.setVisibility(View.VISIBLE);
+                iv_profilepic.setImageBitmap(BitmapFactory
+                        .decodeFile(imgDecodableString));
+
+            } else {
+                Toast.makeText(getActivity(), "You haven't picked Image",
```

Comment:
Use `Toaster` library that we have in utils.

Reply:
done 😃